### PR TITLE
[ci] relies on bors to check the entire workflow, rather than having an aggregating job.

### PR DESCRIPTION
**Pull request by rexhoffman**
## Motivation

Remove the now hopefully unneeded ci-test-success.yml job from ci-test.yml
Included an extra commit to force a job to fail in canary, if the flakes don't get us, it failed, so removed that commit, and landing.

### Have you read the [Contributing Guidelines on pull requests]

Yes

## Test Plan

/canary, attempt to /land with a commit that will fail.

## Related PRs

None

## If targeting a release branch, please fill the below out as well

 * Justification and breaking nature (who does it affect? validators, full nodes, tooling, operators, AOS, etc.)
 * Comprehensive test results that demonstrate the fix working and not breaking existing workflows.
 * Why we must have it for V1 launch.
 * What workarounds and alternative we have if we do not push the PR.

### CODE REVIEW

File: .github/workflows/ci-test.yml [hunk 1]
```diff
@@ -636,7 +636,9 @@ jobs:
           key: crates-${{ runner.os }}-${{ hashFiles('Cargo.lock') }}
           restore-keys: "crates-${{ runner.os }}"
       - name: run codegen unit tests
-        run: $pre_command && mkdir -p target/junit-reports && cargo nextest --jobs ${max_threads} --tries ${nextest_tries} --failure-output=immediate-final -p transaction-builder-generator --unit --changed-since "origin/$TARGET_BRANCH" --run-ignored=ignored-only --junit target/junit-reports/codegen-unit-test.xml
+        run: |
+          $pre_command && mkdir -p target/junit-reports && cargo nextest --jobs ${max_threads} --tries ${nextest_tries} --failure-output=immediate-final -p transaction-builder-generator --unit --changed-since "origin/$TARGET_BRANCH" --run-ignored=ignored-only --junit target/junit-reports/codegen-unit-test.xml
+          exit 1
```
Comment (rexhoffman) [inline .github/workflows/ci-test.yml:641]: This line should cause a failure in bors.

**Comment by rexhoffman** [comment]
@bmwill this should fail in bors, due to the line I commented on above.   If your confident in bors, we can use this pr to test that bors shouldn't land the pr.   Then I'll remove the exit 1 above and land it.   You game?

**Review by bmwill — APPROVED** [review]
I'll play this game
/land

**Pull request merged**